### DISCUSSION TITLE
Do I need to run migrations through keystone?

### DISCUSSION THREAD

**Discussion by simonmcallister0210** [keystonejs/keystone#7448] [Questions] [answered]
Migrations are me trouble at the moment. Would it be a problem to just upload the migration files somewhere, then say download them to a bastion instance and run the migrations for there? Or do they have to be run in the context of the keystone project? (i.e. with the rest of the code with the node_modules and `keystone prisma ...` command?)

**Comment by dcousens** [comment, marked as answer]
[`keystone prisma migrate deploy`](
https://github.com/keystonejs/keystone/blob/3b1c2fe967cbcc1762dfc02e7c935f24c8f57514/packages/core/src/scripts/prisma.ts#L9-L25
) will read your `keystone.ts`, and subsequently call `validateCommittedArtifacts` and `generateNodeModulesArtifacts`; these functions verify things like if `Your Prisma and GraphQL schemas are not up to date` by comparing it against your existing generated schema(s).

For now, these functions are using the source, not the built output; so they need to be run in the context of your source code.
However, the arguments provided to the `keystone prisma...` call (and the database configuration from your Keystone configuration) are passed directly to `prisma` through `execa`,

If you are deploying non-interactively - and you don't want to provide your source or the rest of the Keystone project - you can use `prisma migrate deploy` yourself directly - that will omit this Keystone-aware functionality and only run the Prisma migrations against a `migrations/` directory.

You will need to provide Prisma the information it needs,  such as your `DATABASE_URL` and the Prisma schema;  but it should otherwise be compatible.
